[20.09] python-miio: fix

flyfloh opened the pull request:
Backport of #102183. 

###### Motivation for this change

Fix the python-miio package for the 20.09 relase as well.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

flyfloh committed:
python-miio: fix

(cherry picked from commit 79e81e840415651f4470942f86f19c3fd254efab)

jonringer reviewed (commented):
When backporting changes, please follow https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#backporting-changes.

Namely, you should be doing `git cherry-pick -x <rev>` from a commit that has already landed in master. If the branches have diverged, you may alter the commit or add another commit to ensure that the package is able to still evaluate and build

flyfloh commented:
Oh, I see. Sorry for missing that. Fixed now.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 102210` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.python-miio</li>
    <li>python38Packages.python-miio</li>
  </ul>
</details>

SuperSandro2000 commented:
Result of `nixpkgs-review pr 102210` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python37Packages.python-miio</li>
    <li>python38Packages.python-miio</li>
  </ul>
</details>

jonringer commented:
please rebase on the latest `release-20.09`, and remove the "broken = true;"

```
$ nix-build -A python3Packages.python-miio
error: Package ‘python3.8-python-miio-0.5.3’ in /home/jon/projects/nixpkgs/pkgs/development/python-modules/python-miio/default.nix:34 is marked as broken, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

(use '--show-trace' to show detailed location information)
```

flyfloh committed:
python-miio: unmark as broken

flyfloh commented:
I've rebased and added a commit that removes the 'broken = true'. Is is okay to add the commit here or should I have that merged to master and then include it here?

jonringer commented:
> I've rebased and added a commit that removes the 'broken = true'. Is is okay to add the commit here or should I have that merged to master and then include it here?

Since this doesn't reflect master, yes, it's perfectly find to add another commit

jonringer reviewed (approved):
LGTM
```
https://github.com/NixOS/nixpkgs/pull/102210
2 packages built:
python37Packages.python-miio python38Packages.python-miio
```
